cmd/snap/keymgr.go

```diff
@@ -47,3 +51,49 @@ func getKeypairManager() (KeypairManager, error) {
 	keypairMgr := asserts.NewGPGKeypairManager()
 	return keypairMgr, nil
 }
+
+type takingPassKeyGen interface {
+	Generate(passphrase string, keyName string) error
+}
+
+type ownSecuringKeyGen interface {
+	Generate(keyName string) error
+}
+
+func generateKey(keypairMgr KeypairManager, keyName string) error {
+	switch keyGen := keypairMgr.(type) {
+	case takingPassKeyGen:
+		return takePassGenKey(keyGen, keyName)
+	case ownSecuringKeyGen:
+		err := keyGen.Generate(keyName)
+		if _, ok := err.(*asserts.ExternalUnsupportedOpError); ok {
+			return fmt.Errorf(i18n.G("cannot generate external keypair manager key via snap command, use the appropriate external procedure to create a 4096-bit RSA key under the name/label %q"), keyName)
+		}
+		return err
+	default:
+		return fmt.Errorf("internal error: unsupported keypair manager %T", keypairMgr)
+	}
+}
+
+func takePassGenKey(keyGen takingPassKeyGen, keyName string) error {
+	fmt.Fprint(Stdout, i18n.G("Passphrase: "))
+	passphrase, err := terminal.ReadPassword(0)
+	fmt.Fprint(Stdout, "\n")
+	if err != nil {
+		return err
+	}
+	fmt.Fprint(Stdout, i18n.G("Confirm passphrase: "))
+	confirmPassphrase, err := terminal.ReadPassword(0)
+	fmt.Fprint(Stdout, "\n")
+	if err != nil {
+		return err
+	}
+	if string(passphrase) != string(confirmPassphrase) {
+		return errors.New(i18n.G("passphrases do not match"))
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
Not needed?

Reply:
thanks for catching this, it was in the original code that moved here from cmd_create_key.go